components/service-operator/apis/storage/v1beta1/s3_types.go

```diff
@@ -197,6 +198,18 @@ func (s *S3Bucket) GetStackTemplate() *cloudformation.Template {
 		"Value":       cloudformation.Ref(S3BucketResourceName),
 	}
 
+	template.Outputs[S3BucketURL] = map[string]interface{}{
+		"Description": "Bucket URL to be returned to the user.",
+		"Value": cloudformation.Join(
+			"",
+			[]string{
+				"https://",
+				cloudformation.Ref(S3BucketURL),
+				".s3.eu-west-2.amazonaws.com",
```

Comment:
I'm just a little bit hesitant about putting the logic for constructing the URL into the CloudFormation template instead of the operator code, because that makes it harder to test (and indeed you don't have a test for the actual values here).

I don't know the svcop code well enough to know but: is it possible to do the string formatting in code instead of cloudformation?